pkg/extend/extend.go

```diff
@@ -87,6 +88,20 @@ func extend(d, fsType string) error {
 		log.Printf("No free space on device to extend partition")
 		return nil
 	}
+	if f.PartitionTable.Label == "dos" {
+		out, err := exec.Command("blockdev", "--getsz", d).Output()
```

Comment:
It might be easier to use this https://github.com/moby/moby/blob/master/pkg/devicemapper/ioctl.go
```golang
func ioctlBlkGetSize64(fd uintptr) (int64, error) {
	var size int64
	if _, _, err := unix.Syscall(unix.SYS_IOCTL, fd, BlkGetSize64, uintptr(unsafe.Pointer(&size))); err != 0 {
		return 0, err
	}
	return size, nil
}
```

Reply:
I guess I should also changes sfdisk json structure to use int64 ? https://github.com/linuxkit/linuxkit/blob/master/pkg/extend/extend.go#L43

Reply:
@justincormack should I make this change ? I can open a separate follow-up PR to change the two calls to blockdev by ioctl.

Reply:
Yes to both of those please.

Reply:
It is probably easier to do it in one PR.